cmd/snap/cmd_list.go

```diff
@@ -117,19 +122,26 @@ func (x *cmdList) Execute(args []string) error {
 	esc := x.getEscapes()
 	w := tabWriter()
 
+	var optSize string
+	if x.Size {
+	    optSize = fmt.Sprintf("\t%s", i18n.G("Size"))
+	}
+	
 	// TRANSLATORS: the %s is to insert a filler escape sequence (please keep it flush to the column header, with no extra spaces)
-	fmt.Fprintf(w, i18n.G("Name\tVersion\tRev\tTracking\tPublisher%s\tNotes\n"), fillerPublisher(esc))
+	fmt.Fprintf(w, i18n.G("Name\tVersion\tRev\tTracking%s\tPublisher%s\tNotes\n"), optSize, fillerPublisher(esc))
 
 	for _, snap := range snaps {
 		// doing it this way because otherwise it's a sea of %s\t%s\t%s
-		line := []string{
-			snap.Name,
-			fmtVersion(snap.Version),
-			snap.Revision.String(),
-			fmtChannel(snap.TrackingChannel),
-			shortPublisher(esc, snap.Publisher),
-			NotesFromLocal(snap).String(),
+		line := []string{}
+		line = append(line, snap.Name)
+		line = append(line, fmtVersion(snap.Version))
+		line = append(line, snap.Revision.String())
+		line = append(line, fmtChannel(snap.TrackingChannel))
+		if x.Size {
+			line = append(line, strutil.SizeToStr(snap.InstalledSize))
 		}
+		line = append(line, shortPublisher(esc, snap.Publisher))
```

Comment:
a different approach would be to print the size in alternative to publisher (as the resulting line kind of matches the channel lines snap info) and the list as-is is already quite wide

Reply:
It would be unexpected to me if a column was removed when passing `--size`.

Another method would be a `--columns` argument in which you can choose what is displayed. It could default to `version,revision,tracking,publisher,notes` and `--size` could then be an alias to `version,revision,tracking,size,notes`.